pkg/operator/clustermemberremovalcontroller/clustermemberremovalcontroller.go

```diff
@@ -390,6 +410,40 @@ func (c *clusterMemberRemovalController) attemptToRemoveMemberFor(ctx context.Co
 	return removed, errs
 }
 
+func (c *clusterMemberRemovalController) getHealthyLiveVotingMembers(ctx context.Context) ([]*etcdserverpb.Member, error) {
```

Comment:
> I can add the unit tests, but they do nothing but calling the client :)

yes, but the output matters and should be asserted.

> I actually wanted to do that, but the issue is that I need to change the etcdcli and move some helper methods to ceohelpers .. I just did not want to make many changes for bug fix :)

That's okay, it's not really a bugfix is it? or do you want to backport it to 4.11?

Reply:
I think so, or wdyt @hasbro17 , @JoelSpeed ?

Reply:
It is technically a bugfix (since I didn't intend this behavior in the original implementation) but there isn't a pressing need to backport this back to 4.11 since it's not like a customer nor CPMS is exercising this particular scenario. This is mostly just to bring this in line with the original design.
I'm okay with deduplicating the helper functions though.

Reply:
@tjungblu 

> actually with getAllLiveVotingMembers - do you think we should put both functions into the etcdClient directly? Seems like a common functionality and duplicate code...

I can 

```
func getVotingMembers(members []*etcdserverpb.Member) []*etcdserverpb.Member {
	var votingMembers []*etcdserverpb.Member
	for _, member := range members {
		if member.IsLearner {
			continue
		}
		votingMembers = append(votingMembers, member)
	}
	return votingMembers
}
```

but I am confused between `ceohelpers` and `etcdcli` ?

Reply:
if I keep both helper functions in `clustermemberremovalcontroller` I can refactor and use this helper, and add tests

Reply:
Just another example which we should refactor with that is:

https://github.com/openshift/cluster-etcd-operator/blob/master/pkg/operator/defragcontroller/defragcontroller.go#L124-L135

So how about we add a new method to `etcdcli.MemberLister`  like:

```
type MemberLister interface {
        // MemberList lists all members in a cluster 
	MemberList(ctx context.Context) ([]*etcdserverpb.Member, error)
        // VotingMemberList lists all voting members in a cluster (ie are not learner)
	VotingMemberList(ctx context.Context) ([]*etcdserverpb.Member, error)
}
```
That way it's well contained and you can test it there along with the existing MemberList.

Reply:
I reckon we could also add equivalent methods to the `UnhealthyMemberLister` and the `HealthyMemberLister`, so we don't have to write filter methods for those all the time.

Reply:
sounds like a good plan 👍🏽  .. i am on it